CommonKit/Sources/CommonKit/Helpers/AdamantSecureStorage.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  AdamantSecureStorage.swift
+//
+//
+//  Created by Stanislav Jelezoglo on 02.08.2024.
+//
+
+import Foundation
+
+final class AdamantSecureStorage: SecureStorageProtocol {
+    let tag = "com.adamant.keys.id".data(using: .utf8)!
+    
+    func getPrivateKey() -> SecKey? {
+        guard let existingKey = loadPrivateKey() else {
+            return createAndStorePrivateKey()
+        }
+        
+        return existingKey
+    }
+    
+    func getPublicKey(privateKey: SecKey) -> SecKey? {
+        SecKeyCopyPublicKey(privateKey)
+    }
+    
+    func encrypt(data: Data, publicKey: SecKey) -> Data? {
+        guard let encryptedData = SecKeyCreateEncryptedData(
+            publicKey,
+            .eciesEncryptionCofactorX963SHA256AESGCM,
+            data as CFData,
+            nil
+        ) else {
+            return nil
+        }
+        
+        return encryptedData as Data
+    }
+    
+    func decrypt(data: Data, privateKey: SecKey) -> Data? {
+        guard let decryptedData = SecKeyCreateDecryptedData(
+            privateKey,
+            .eciesEncryptionCofactorX963SHA256AESGCM,
+            data as CFData,
+            nil
+        ) else {
+            return nil
+        }
+        
+        return decryptedData as Data
+    }
+}
+
+private extension AdamantSecureStorage {
+    func loadPrivateKey() -> SecKey? {
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassKey,
+            kSecAttrApplicationTag as String: tag,
+            kSecAttrKeyType as String: kSecAttrKeyTypeEC,
+            kSecReturnRef as String: true
+        ]
+        
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(query as CFDictionary, &item)
+        
+        if status == errSecSuccess {
+            return (item as! SecKey)
+        }
+        
+        return nil
```

Comment:
```
return status == errSecSuccess
     ? item as? SecKey
     : nil
```

Reply:
the same

Reply:
Yes, it's same, but we always have to write shorter code